src/lib/fetch.ts

```diff
@@ -0,0 +1,110 @@
+import {
+    Provider,
+    ResourceTimingEntry,
+    ResourceTimingEntryValidationPredicate,
+    TestConfiguration,
+    TestResultBundle,
+    TestSetupResult,
+} from "../@types"
+import { asyncGetEntry } from "./resourceTiming"
+import { Test } from "./test"
+
+export type FetchConfiguration = TestConfiguration & {
+    performanceTimingObserverTimeout?: number
+}
+
+/**
+ * The default function used to validate Resource Timing entries.
+ * @param e The entry to validate.
+ */
+const defaultIsValidEntryFunc: ResourceTimingEntryValidationPredicate = (e) =>
+    e.requestStart !== 0 && e.connectStart !== e.connectEnd
+
+/**
+ * Class representing a basic "fetch" test. Along with its parent class
+ * {@link Test}, the class provides a number of hooks enabling providers to
+ * control certain implementation details supporting their use case.
```

Comment:
Added some more detail to docs as this is the main test type most providers will use.
```suggestion
 * Class representing a basic "fetch" test. It fetches a supplied resource from the 
 * network and collects the corresponding ResourceTimingEntry from the 
 * Performance timeline. 
 *
 * Along with its parent class {@link Test}, the class provides a number of hooks 
 * enabling providers to control certain implementation details supporting their 
 * use case.
```